extension/src/extension/Extension.ts

```diff
@@ -209,14 +219,22 @@ export class Extension {
   private handleEditorEvents(): void {
     this.context.subscriptions.push(
       vscode.workspace.onDidChangeConfiguration(
-        (event: vscode.ConfigurationChangeEvent) => {
+        async (event: vscode.ConfigurationChangeEvent) => {
           if (
             event.affectsConfiguration('java.home') &&
             this.server.isReady()
           ) {
             this.server.restart();
           }
-          if (event.affectsConfiguration('gradle.javaDebug')) {
+          if (event.affectsConfiguration('gradle.nestedProjects')) {
+            this.rootProjectsStore.clear();
+            this.gradleTaskProvider.clearTasksCache();
```

Comment:
once https://github.com/badsyntax/vscode-gradle/pull/493 is merged we should use this.reloadTasks()